### ISSUE TITLE
Class Loader Error in Development Environment

### ISSUE THREAD

**Issue by gamerYToff** [CleanroomMC/MixinBooter#94]
It was probably a mistake on my part during installation, but for some reason, IFMLLoadingPlugin is loaded by the sun.misc.Launcher$AppClassLoader, while MixinBootPlugin is loaded by the net.minecraft.launchwrapper.LaunchClassLoader. Due to this difference in class loaders, from the JVM's perspective, MixinBooterPlugin does not implement IFMLLoadingPlugin, which causes a ClassCastException in the loadCoreMod method of the CoreModManager class. As a result, Forge interprets the mod as not being a core mod, and therefore it doesn't load in the development environment. I managed to fix this by modifying how this class handles the casting, but I still want to understand why this error occurred, if anyone else has faced it, and what I likely did wrong.

**Comment by Rongmario** [comment]
Would like to see the log, are you making a mod with MixinBooter as dependency or are you purely trying to run the mod?

**Comment by gamerYToff** [comment]
> Would like to see the log, are you making a mod with MixinBooter as dependency or are you purely trying to run the mod?

Well, as I was looking through the repositories from this group, I ended up finding the build template, and with that, I was able to fix this problem and a few bugs. I’m not exactly sure why that was happening, but after seeing this build template, I have some ideas. It's a bit embarrassing, but my build.gradle was terrible—lots of useless stuff, issues, and workarounds. Anyway, thank you very much for your attention; I didn't expect anyone to actually answer both questions. Hugs, and thanks again.
